### PR TITLE
Move currency list from JS to Clojure

### PR DESCRIPTION
**Pull request by noahmoss**
This PR moves the list of currencies from JS to a `.cljc` file so that it can be accessed from the frontend and backend. This is because I'll need it on the backend for formatting currencies in XLSX exports.

I've changed the data structure to a nested vector in order to preserve the order of currencies, and I've updated the usage on the frontend accordingly.

Qs:
* Does the filepath/namespace make sense? Not sure how we're organizing shared code.
* Do I need to modify shadow.cljs with the new namespace added to `:entries`? It's the only way I could get the build to work, but I see other files in `frontend/src/cljs` that aren't listed there.

### CODE REVIEW

File: shared/src/metabase/shared/util/currency.cljc [hunk 1]
```diff
@@ -0,0 +1,851 @@
...
+          :decimal_digits 2,
+          :rounding 0,
+          :code "ZAR",
+          :name_plural "South African rand"}],
+   [:ZMK {:symbol "ZK",
+          :name "Zambian Kwacha",
+          :symbol_native "ZK",
+          :decimal_digits 0,
+          :rounding 0,
+          :code "ZMK",
```
Comment (howonlee) [inline shared/src/metabase/shared/util/currency.cljc:843]: code is ZMW now

File: shared/src/metabase/shared/util/currency.cljc [hunk 2]
```diff
@@ -0,0 +1,851 @@
...
+          :name_plural "Vietnamese dong"}],
+   [:XAF {:symbol "FCFA",
+          :name "CFA Franc BEAC",
+          :symbol_native "FCFA",
+          :decimal_digits 0,
+          :rounding 0,
+          :code "XAF",
+          :name_plural "CFA francs BEAC"}],
+   [:XOF {:symbol "CFA",
+          :name "CFA Franc BCEAO",
```
Comment (howonlee) [inline shared/src/metabase/shared/util/currency.cljc:818]: not 0 the number, O the letter, BCEAO is acronym

File: shared/src/metabase/shared/util/currency.cljc [hunk 3]
```diff
@@ -0,0 +1,851 @@
...
+          :name_plural "Uruguayan pesos"}],
+   [:UZS {:symbol "UZS",
+          :name "Uzbekistan Som",
+          :symbol_native "UZS",
+          :decimal_digits 0,
+          :rounding 0,
+          :code "UZS",
+          :name_plural "Uzbekistan som"}],
+   [:VEF {:symbol "Bs.F.",
+          :name "Venezuelan Bolívar",
```
Comment (howonlee) [inline shared/src/metabase/shared/util/currency.cljc:797]: the hyperinflation in bolivar led them to get rid of the bs fuerte: it's the bs soberano now

File: shared/src/metabase/shared/util/currency.cljc [hunk 4]
```diff
@@ -0,0 +1,851 @@
...
+          :name_plural "Ugandan shillings"}],
+   [:UYU {:symbol "$U",
+          :name "Uruguayan Peso",
+          :symbol_native "$",
+          :decimal_digits 2,
+          :rounding 0,
+          :code "UYU",
+          :name_plural "Uruguayan pesos"}],
+   [:UZS {:symbol "UZS",
+          :name "Uzbekistan Som",
```
Comment (howonlee) [inline shared/src/metabase/shared/util/currency.cljc:790]: so'm has an apostrophe

File: shared/src/metabase/shared/util/currency.cljc [hunk 5]
```diff
@@ -0,0 +1,851 @@
...
+          :name "Trinidad and Tobago Dollar",
+          :symbol_native "$",
+          :decimal_digits 2,
+          :rounding 0,
+          :code "TTD",
+          :name_plural "Trinidad and Tobago dollars"}],
+   [:TWD {:symbol "NT$",
+          :name "New Taiwan Dollar",
+          :symbol_native "NT$",
+          :decimal_digits 2,
```
Comment (howonlee) [inline shared/src/metabase/shared/util/currency.cljc:757]: new taiwan dollar doesn't have decimals in common usage, it's like 30 ntd to 1 usd

**Comment by noahmoss** [comment]
@howonlee Nice catches. I'll make them in a separate PR so that this one can be focused on the js -> cljs migration. Will make it easier to see where they were done in the git history.

File: shared/src/metabase/shared/util/currency.cljc [hunk 6]
```diff
@@ -0,0 +1,851 @@
...
+   [:SGD {:symbol "S$",
+          :name "Singapore Dollar",
+          :symbol_native "$",
+          :decimal_digits 2,
+          :rounding 0,
+          :code "SGD",
+          :name_plural "Singapore dollars"}],
+   [:SOS {:symbol "Ssh",
+          :name "Somali Shilling",
+          :symbol_native "Ssh",
```
Comment (howonlee) [inline shared/src/metabase/shared/util/currency.cljc:707]: it's Sh.So for Somalian shillings

File: shared/src/metabase/shared/util/currency.cljc [hunk 7]
```diff
@@ -0,0 +1,851 @@
...
+          :symbol_native "FC",
+          :decimal_digits 0,
+          :rounding 0,
+          :code "KMF",
+          :name_plural "Comorian francs"}],
+   [:KRW {:symbol "₩",
+          :name "South Korean Won",
+          :symbol_native "₩",
+          :decimal_digits 0,
+          :rounding 0,
```
Comment (howonlee) [inline shared/src/metabase/shared/util/currency.cljc:429]: krw is rounded to the 10 won. 10 won ~~= 1 cent usd

File: shared/src/metabase/shared/util/currency.cljc [hunk 8]
```diff
@@ -0,0 +1,851 @@
...
+          :code "PHP",
+          :name_plural "Philippine pesos"}],
+   [:PKR {:symbol "PKRs",
+          :name "Pakistani Rupee",
+          :symbol_native "₨",
+          :decimal_digits 0,
+          :rounding 0,
+          :code "PKR",
+          :name_plural "Pakistani rupees"}],
```
Comment (howonlee) [inline shared/src/metabase/shared/util/currency.cljc:627]: one day we need to implement counting by indian and east asian customary values (lakhs/crore, wan/man, etc)

probably not this day but one day

File: shared/src/metabase/shared/util/currency.cljc [hunk 9]
```diff
@@ -0,0 +1,851 @@
...
+          :name "Israeli New Sheqel",
+          :symbol_native "₪",
+          :decimal_digits 2,
+          :rounding 0,
+          :code "ILS",
+          :name_plural "Israeli new sheqels"}],
+   [:INR {:symbol "Rs",
+          :name "Indian Rupee",
+          :symbol_native "টকা",
+          :decimal_digits 2,
```
Comment (howonlee) [inline shared/src/metabase/shared/util/currency.cljc:351]: paisa is very rarely used wrt rupee

File: shared/src/metabase/shared/util/currency.cljc [hunk 10]
```diff
@@ -0,0 +1,851 @@
...
+          :name_plural "Hungarian forints"}],
+   [:IDR {:symbol "Rp",
+          :name "Indonesian Rupiah",
+          :symbol_native "Rp",
+          :decimal_digits 0,
+          :rounding 0,
+          :code "IDR",
+          :name_plural "Indonesian rupiahs"}],
+   [:ILS {:symbol "₪",
+          :name "Israeli New Sheqel",
```
Comment (howonlee) [inline shared/src/metabase/shared/util/currency.cljc:342]: standard romanization is shekel

File: shared/src/metabase/shared/util/currency.cljc [hunk 11]
```diff
@@ -0,0 +1,851 @@
...
+          :name_plural "Swiss francs"}],
+   [:CLP {:symbol "CL$",
+          :name "Chilean Peso",
+          :symbol_native "$",
+          :decimal_digits 0,
+          :rounding 0,
+          :code "CLP",
+          :name_plural "Chilean pesos"}],
+   [:CNY {:symbol "CN¥",
+          :name "Chinese Yuan",
```
Comment (howonlee) [inline shared/src/metabase/shared/util/currency.cljc:181]: official name is renminbi

**Comment by howonlee** [comment]
> @howonlee Nice catches. I'll make them in a separate PR so that this one can be focused on the js -> cljs migration. Will make it easier to see where they were done in the git history.

makes sense. I just went on a nitpicking spree after finding a few bits

**Comment by howonlee** [comment]
I don't know whether it's strictly good to have the NTD rounding change or the KRW rounding change I suggested, because people do use single-digit KRW and NTD cents in stock prices, which I guess people use mb for a lot

**Review by howonlee — APPROVED** [review]
no syntactic problems, found buncha semantic nits but as noah said, good subject for another PR

File: shared/src/metabase/shared/util/currency.cljc (hunk 11 above)
Comment (howonlee) [reply shared/src/metabase/shared/util/currency.cljc:181]: (this is kind of an oversimplication: renminbi is name of currency, yuan is measure of renminbi in actuality. but we have pound sterling as name of the pound, so we should have renminbi as the name)

**Comment by howonlee** [comment]
i see we round chf to 5 cents because they got rid of their penny.  but australia, canada, denmark, finland, hong kong, hungary, israel, malaysia, mexico, new zealand, sweden and new guinea hve removed denominations - but again, the need for graphing 1 cent-equivalent denominations stayed on nearly all of them

**Comment by noahmoss** [comment]
@howonlee We're currently not actually using most of this data for currency formatting. We're just using it for the list of options. See #10785

**Pull request merged**